lib/chef/provider/package/windows.rb

```diff
@@ -38,7 +38,7 @@ class Windows < Chef::Provider::Package
         def define_resource_requirements
           if new_resource.checksum
             requirements.assert(:install) do |a|
-              a.assertion { new_resource.checksum.downcase == checksum(source_location) }
+              a.assertion { new_resource.checksum == checksum(source_location) }
```

Comment:
windows_package already does this coercion: https://github.com/chef/chef/blob/master/lib/chef/resource/windows_package.rb#L140-L142